packages/springer-dropdown/js/dropdown.js

```
@@ -1,43 +1,25 @@
-import {makeArray, getDataOptions} from '@springernature/global-context/js/helpers';
+import {constants} from './constants';
 
 /**
- * Constants
+ * Local Constants
  */
-
-const DATA_COMPONENT = 'data-dropdown';
-
-const ClassName = {
-	DROPDOWN: 'c-dropdown',
-	DROPDOWN_OPEN_MODIFIER: 'c-dropdown--open',
-	MENU: 'c-dropdown__menu',
-	ITEM: 'c-dropdown__item',
-	LINK: 'c-dropdown__link'
-};
-
-const Selector = {
-	DATA_COMPONENT: `[${DATA_COMPONENT}]`,
-	DROPDOWN: `.${ClassName.DROPDOWN}`,
-	MENU: `.${ClassName.MENU}`
+const selector = {
+	DROPDOWN: `.${constants.DROPDOWN}`,
+	MENU: `.${constants.MENU}`
 };
 
-const DefaultOptions = {
-	HIDE_CLASS: 'u-hidden',
+const defaultOptions = {
+	HIDE_CLASS: 'u-hide',
 	HIDE_INITIALLY: true,
 	CLICK_OUTSIDE: true
 };
 
-const DataOptions = {
-	HIDE_CLASS: DATA_COMPONENT + '-hide-class',
-	HIDE_INITIALLY: DATA_COMPONENT + '-hide-initially',
-	CLICK_OUTSIDE: DATA_COMPONENT + '-click-outside'
-};
-
 const Dropdown = class {
 	constructor(button, options = {}) {
-		this._options = Object.assign({}, DefaultOptions, options);
+		this._options = Object.assign({}, defaultOptions, options);
 		this._buttonEl = button;
-		this._dropdownEl = this._buttonEl.closest(Selector.DROPDOWN);
-		this._menuEl = this._dropdownEl.querySelector(Selector.MENU);
+		this._dropdownEl = this._buttonEl.closest(selector.DROPDOWN);
+		this._menuEl = this._dropdownEl.querySelector(selector.MENU);
```

Is the intended behavior that this will throw a JS error if the HTML isn't correct? When accessing `this._dropdownEl.classList` etc it'll bottle out the current script.

---

That was my thought process, it completely doesn't work if the HTML is incorrect so I thought I'd throw.  I can return silently though...

---

I'd throw an error message in the constructor if those elements wheren't found. The constructor should bring the instance into a usable state, and without those elements the instance can never be stable.
Without the explicit throw it will be confusing why the call to `.init` is throwing `undefined does not have property X` sort of messages

---

Oh I misunderstood.  You mean safety check the `buttonEl`